### PR TITLE
Scheduler GraphQL Migration

### PR DESCRIPTION
**Pull request by dhuangg**
After much struggle and strife, was able to get basic date filtering and navigation complete. This is to address #649

### CODE REVIEW

**Review by CalvinFronda — COMMENTED** [review]
Logging into a parent account. Click on "Courses" > Click on enrolled course. It will break because 

```
Uncaught TypeError: Cannot read property 'user' of undefined
    at identifyCourseHasEnrolledStudent (CourseClass.js:206)
```

removing ``` student: {user}``` fixes the onCompleted method in the query. 

What is this prop need to do that the global ```user``` variable can't achieve?

File: src/components/FeatureViews/Scheduler/SessionView.js [hunk 1]
```diff
@@ -304,33 +304,24 @@ const SessionView = () => {
...
                 </Grid>
-                <Grid item></Grid>
                 <Grid item>
-                    <Button
-                        className='editButton'
+                    <ResponsiveButton
                         color='primary'
                         onClick={handleEditToggle(true)}
-                        variant='outlined'
+                        variant='contained'
```
Comment (CalvinFronda) [inline src/components/FeatureViews/Scheduler/SessionView.js:321]: A parent shouldn't be able to see the Reschedule button. Suggest adding some account validation.

File: src/components/Form/FormUtils.js [hunk 2]
```diff
@@ -257,6 +257,26 @@ export function arr_diff(a1, a2) {
     return diff;
 }
 
+// Function definition with passing two arrays
```
Comment (CalvinFronda) [inline src/components/Form/FormUtils.js:260]: You can make this faster by using .some & .includes 

```array1.some(item => array2.includes(item))``` 

The issue is ```.includes``` & ```.some``` isn't supported on IE.

**Review by CalvinFronda — CHANGES_REQUESTED** [review]

**Comment by dhuangg** [comment]
> Logging into a parent account. Click on "Courses" > Click on enrolled course. It will break because
> 
> ```
> Uncaught TypeError: Cannot read property 'user' of undefined
>     at identifyCourseHasEnrolledStudent (CourseClass.js:206)
> ```
> 
> removing ` student: {user}` fixes the onCompleted method in the query.
> 
> What is this prop need to do that the global `user` variable can't achieve?

This isn't to compare with the authenticated user (the global user var), this is to see whether or not a parent's student is enrolled in a class. This error may be from the query failing. I'm not getting this error so we'll need to look into this a bit more

File: src/components/Form/FormUtils.js (hunk 2 above)
Comment (dhuangg) [reply src/components/Form/FormUtils.js:260]: i knew there was a smarter way to do this xD. thanks!

File: src/components/FeatureViews/Scheduler/Scheduler.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ const EventPopoverWrapper = ({ children, popover }) => {
     };
 
     const handlePopoverOpen = (event) => {
+        console.log(event.currentTarget);
```
Comment (CalvinFronda) [inline src/components/FeatureViews/Scheduler/Scheduler.js:44]: remove log

**Review by juicekey — COMMENTED** [review]
Looks good to me

**Pull request merged**